### PR TITLE
MAP-1095 add endpoint to get non-res locations buy usage type

### PR DESCRIPTION
**Pull request by GurnankCheema**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/locationsinsideprison/jpa/repository/LocationRepository.kt [hunk 1]
```diff
@@ -18,3 +20,9 @@ interface LocationRepository : JpaRepository<Location, UUID> {
   @Query("select l from Location l where concat(l.prisonId,'-',l.pathHierarchy) IN (:keys)")
   fun findAllByKeys(keys: List<String>): List<Location>
 }
+
+@Repository
+interface NonResidentialLocationRepository : JpaRepository<NonResidentialLocation, UUID> {
+  @Query("select distinct nrl from NonResidentialLocation nrl, Location l left join nrl.nonResidentialUsages u where u.usageType = :usageType and l.prisonId = :prisonId")
```
Comment (Mjwillis) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/locationsinsideprison/jpa/repository/LocationRepository.kt:26]: This doesn't quite right - it doesn't need a left join or an additional Location
```kotlin
@Query("select distinct nrl from NonResidentialLocation nrl join nrl.nonResidentialUsages u where u.usageType = :usageType and l.prisonId = :prisonId")
```

**Review by Mjwillis — CHANGES_REQUESTED** [review]
JQL query needs minor change

File: src/main/kotlin/uk/gov/justice/digital/hmpps/locationsinsideprison/jpa/repository/LocationRepository.kt (hunk 1 above)
Comment (GurnankCheema) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/locationsinsideprison/jpa/repository/LocationRepository.kt]: refactored

**Review by Mjwillis — APPROVED** [review]
LGTM

**Pull request merged**